jxl/src/frame/modular/transforms.rs

```diff
@@ -463,11 +463,32 @@ pub fn make_grids(
     // Create grids.
     for g in buffer_info.iter_mut() {
         g.buffer_grid = get_grid_indices(g.grid_kind)
-            .map(|_| ModularBuffer {
-                data: None,
-                auxiliary_data: None,
-                remaining_uses: 0,
-                used_by_transforms: vec![],
+            .map(|(x, y)| {
+                let chan_size = g.info.size;
+                let size = match g.grid_kind {
+                    ModularGridKind::None => chan_size,
+                    ModularGridKind::Lf => {
+                        let dx = frame_header.lf_group_dim() >> g.info.shift.unwrap().0;
+                        let bx = x as usize * dx;
+                        let dy = frame_header.lf_group_dim() >> g.info.shift.unwrap().1;
+                        let by = y as usize * dy;
```

Comment:
I suppose one could write a function like
```
fn get_group_dim_for_grid_kind(grid_kind: ModularGridKind, frame_header: &FrameHeader) -> usize {
    match grid_kind {
        ModularGridKind::Lf => frame_header.lf_group_dim(),
        ModularGridKind::Hf => frame_header.group_dim(),
        _ => panic!()
    }
}
```
and then avoid having two almost identical branches here. Not sure if this makes is less readable....

Reply:
Did something slightly different, PTAL

Reply:
nice!